### PR TITLE
set up e2e

### PR DESCRIPTION
**Pull request by dmsolr**

### CODE REVIEW

**Comment by wu-sheng** [comment]
Please merge the master, since we haven't opened the branch sync. But I have submitted the JIRA.

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+name: E2E
+
+on:
+  pull_request:
+  push:
+    branches: 
+      - master
+    tags:
+      - 'v*'
+
```
Comment (rainbend) [inline .github/workflows/e2e.yaml]:
```suggestion
```

**Comment by rainbend** [comment]
We should move the `mock-collector` and `validator` code to a new repo, other agent repo can also be used, @dmsolr what do you think?

**Comment by wu-sheng** [comment]
> We should move the mock-collector and validator code to a new repo, other agent repo can also be used

If you intend to do this, we will need a proposal and a release process(or we indicate no release for that repo) for this in the mail list. This proposal should be approved(default yes) from the PMC.

**Comment by rainbend** [comment]
> If you intend to do this

Yes. otherwise we have to maintain code in multiple repos(at least two).

**Comment by wu-sheng** [comment]
> Yes. otherwise we have to maintain code in multiple repos(at least two).

Yes, and NO. Currently, the main repo codes don't support reporting through HTTP.

**Comment by wu-sheng** [comment]
@dmsolr I am cancelling the action, as it runs near 1h and no response.

**Comment by dmsolr** [comment]
Hi @wu-sheng 
This error message looks like a program issue.
```
18:05:38.065 openresty-with-skywalking> 2020/02/26 18:05:38 [error] 6#6: *10 lua entry thread aborted: runtime error: /skywalking-nginx-lua/lib/skywalking/util.lua:106: attempt to index local 'id' (a nil value)
18:05:38.065 openresty-with-skywalking> stack traceback:
18:05:38.065 openresty-with-skywalking> coroutine 0:
18:05:38.065 openresty-with-skywalking> 	/skywalking-nginx-lua/lib/skywalking/util.lua: in function 'id2String'
18:05:38.065 openresty-with-skywalking> 	/skywalking-nginx-lua/lib/skywalking/segment_ref.lua:108: in function 'serialize'
18:05:38.065 openresty-with-skywalking> 	/skywalking-nginx-lua/lib/skywalking/span.lua:139: in function 'createExitSpan'
18:05:38.065 openresty-with-skywalking> 	/skywalking-nginx-lua/lib/skywalking/tracer.lua:62: in function 'startBackendTimer'
18:05:38.065 openresty-with-skywalking> 	rewrite_by_lua(nginx.conf:74):2: in main chunk, client: 127.0.0.1, server: , request: "GET /tier2/lb HTTP/1.0", host: "127.0.0.1:8080"
```

**Review by wu-sheng — APPROVED** [review]
LGTM. Let's merge this first to make sure the codes are good of every PR. @dmsolr @arugal If you want the new `skywalking-agent-test-tool` repo, please start from the mail list proposal officially.

**Pull request merged**